test2code/src/main/kotlin/com/epam/drill/test2code/coverage/RetentionQueue.kt

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2020 - 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.drill.test2code.coverage
+
+import mu.*
+import java.math.*
+import java.util.*
+import java.util.concurrent.*
+
+
+interface RetentionQueue {
+    fun addAll(data: Sequence<ByteArray>)
+    fun flush(): Sequence<ByteArray>
+}
+
+class InMemoryRetentionQueue(
+    private var queue: Queue<ByteArray> = ConcurrentLinkedQueue(),
+    private val totalSizeByteLimit: BigInteger
+) : RetentionQueue {
+
+    init {
+        require(totalSizeByteLimit >= BigInteger.ZERO) {
+            "Total size byte limit must be a positive value"
+        }
```

Comment:
@bodyangug add a log to print warning when `totalSizeByteLimit` is set to `0`
It's a valid value, but might cause problems if left unnoticed.
`agent parameter "coverageRetentionLimit" limit is set to 0 (zero). Coverage won't be retained in case Drill4J Admin Backend is unavailable."`

Reply:
done